ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -61,15 +102,122 @@ export class ImportedDocuments extends React.Component {
         });
       } else {
         this.setState({
-          data: response.data,
           isLoading: true,
+          data: response.data,
         });
       }
+      this.calculateMode();
```

Comment:
I understand the need of updating the state with the `filteredRecords`, but we don't have to recalculate the state for `mode` and `status` and assign a label.
Here I would:
```
this.setState({
          isLoading: true,
          data: response.data, 
          mode: response.data.mode,
          status: response.data.status
        });
```
Do you need to store the all `data` in the state? Please remove it if not needed.

Then, in the render method:
```
render() {
    const { mode, status, ...} = this.state; 
    const modeCmp = this.getModeCmp(mode);
    const statusCmp = this.getStatusCmp(status);
}
```

Reply:
was the recalculating filters and statistic for all the data (even the previously fetched already) addressed ?